gen_schema.py

```diff
@@ -0,0 +1,41 @@
+from typing import Annotated, Any, Sequence
```

Comment:
do we want to keep this file at the root of the project? Or should we move it to the kpops folder?

Reply:
ideally this should be a kpops command imo, but I would like to do it in a followup when we also add the documentation for it. for now I would simply make it a standalone script if that's ok

Reply:
Yeah, this is totally fine for now. But I don't think this should be part of `kpops` command. IMO the user should have the generated schema and import it. In other words, we should create a CI job that runs after each PR and updates the schemas. In that way, we have the most updated schemas, and the users can use them.

Reply:
Yes, sure, I agree we should provide a default schema. Exposing this as a command would have the advantage that the user can generate their own schema including all the custom components inside their `COMPONENTS_MODULE`. I think that would be the next logical step and improve the user experience. Otherwise they will likely get validation errors when they use custom components in their `pipeline.yaml`.

Reply:
Agree! Please provide a follow-up issue on this!

Reply:
I've opened #71 and #72 for this